clients/java/src/main/java/io/camunda/client/impl/statistics/filter/ProcessDefinitionStatisticsFilterImpl.java

```diff
@@ -0,0 +1,200 @@
+    return this;
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl endDate(final OffsetDateTime endDate) {
+    endDate(b -> b.eq(endDate));
+    return this;
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl endDate(final Consumer<DateTimeProperty> fn) {
+    final DateTimeProperty property = new DateTimePropertyImpl();
+    fn.accept(property);
+    filter.setEndDate(RequestMapper.toProtocolObject(property.build()));
+    return this;
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl state(final ProcessInstanceState state) {
+    return state(b -> b.eq(state));
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl state(
+      final Consumer<ProcessInstanceStateProperty> fn) {
+    final ProcessInstanceStateProperty property = new ProcessInstanceStatePropertyImpl();
+    fn.accept(property);
+    filter.setState(RequestMapper.toProtocolObject(property.build()));
+    return this;
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl hasIncident(final Boolean hasIncident) {
+    filter.hasIncident(hasIncident);
+    return this;
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl tenantId(final String tenantId) {
+    tenantId(b -> b.eq(tenantId));
+    return this;
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl tenantId(final Consumer<StringProperty> fn) {
+    final StringProperty property = new StringPropertyImpl();
+    fn.accept(property);
+    filter.setTenantId(RequestMapper.toProtocolObject(property.build()));
+    return this;
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl variables(
+      final List<ProcessInstanceVariableFilterRequest> variableValueFilters) {
+    if (variableValueFilters != null) {
+      variableValueFilters.forEach(v -> variableValueNullCheck(v.getValue()));
+    }
+    filter.setVariables(RequestMapper.toProtocolList(variableValueFilters));
+    return this;
+  }
+
+  @Override
+  public ProcessDefinitionStatisticsFilterImpl variables(
+      final Map<String, Object> variableValueFilters) {
+    if (variableValueFilters != null && !variableValueFilters.isEmpty()) {
+      final List<ProcessInstanceVariableFilterRequest> variableFilters =
+          variableValueFilters.entrySet().stream()
+              .map(
+                  entry -> {
+                    variableValueNullCheck(entry.getValue());
+                    final ProcessInstanceVariableFilterRequest request =
+                        new ProcessInstanceVariableFilterRequest();
+                    request.setName(entry.getKey());
+                    final StringProperty property = new StringPropertyImpl();
+                    property.eq(entry.getValue().toString());
+                    request.setValue(property.build());
+                    return request;
+                  })
+              .collect(Collectors.toList());
+      filter.setVariables(RequestMapper.toProtocolList(variableFilters));
+    }
```

Comment:
❓ Is this duplicated now with the process instance search request? I could imagine we do the same thing there? If so, let's extract this into the request mapper or so.

Reply:
Yes, it's exactly the same.

Reply:
I moved the snippet to the `RequestMapper` and found another duplication in `UserTaskFilterImpl`, simplified that too.